metrics/collectors/images/images_collector.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2021 Canonical Ltd
+
+import datetime
+import re
+import subprocess
+import tempfile
+
+from launchpadlib.launchpad import Launchpad
+from metrics.lib.basemetric import Metric
+
+RSYNC_SERVER_REQUESTS = [
+    "rsync://cdimage.ubuntu.com/cdimage/daily*/*/*.",
+    "rsync://cdimage.ubuntu.com/cdimage/*/daily*/*/*.",
+    "rsync://cdimage.ubuntu.com/cdimage/*/*/daily*/*/*.",
+]
+IMAGE_FORMATS = ["iso", "img.xz"]
+
+
+class ImagesMetrics(Metric):
+    def __init__(self, dry_run=False, verbose=False):
+        super().__init__(dry_run, verbose)
+
+        self.lp = Launchpad.login_anonymously(
+            "metrics",
+            "production",
+            launchpadlib_dir=tempfile.mkdtemp(),
+            version="devel",
+        )
+        self.ubuntu = self.lp.distributions["ubuntu"]
+        self.active_series = {s.name: s for s in self.ubuntu.series if s.active}
+        self.date_now = datetime.datetime.now()
+
+    def rsync_list_images(self):
+        rsync = ""
+        for url in RSYNC_SERVER_REQUESTS:
+            for img in IMAGE_FORMATS:
+                try:
+                    self.log.debug("Rsync listing %s" % url + img)
```

Comment:
nitpick:

```suggestion
                    self.log.debug("Rsync listing %s%s", url, img)
```

Reply:
Alright, to me they seem equivalent options but if you are a preference let's go with that!